zcash_client_sqlite/src/lib.rs

```diff
@@ -665,35 +662,31 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
         d_tx: DecryptedTransaction<AccountId>,
     ) -> Result<(), Self::Error> {
         self.transactionally(|wdb| {
-            let tx_ref = wallet::put_tx_data(wdb.conn.0, d_tx.tx, None, None)?;
+            let tx_ref = wallet::put_tx_data(wdb.conn.0, d_tx.tx(), None, None)?;
 
             let mut spending_account_id: Option<AccountId> = None;
-            for output in d_tx.sapling_outputs {
-                match output.transfer_type {
+            for output in d_tx.sapling_outputs() {
+                match output.transfer_type() {
                     TransferType::Outgoing | TransferType::WalletInternal => {
-                        let value = output.note.value();
-                        let recipient = if output.transfer_type == TransferType::Outgoing {
-                            Recipient::Sapling(output.note.recipient())
+                        let recipient = if output.transfer_type() == TransferType::Outgoing {
+                            //TODO: Recover the UA, if possible.
```

Comment:
If you're going to do something clever to build a UA from a transaction recipient, please only do so into some new Option field for that purpose. At least, there should be a straightforward way to determine the _actual_ recipient address (read: _one_ receiver) for a given output as a consumer of this API.